### ISSUE TITLE
API_KEY not consistently recognized by CLI

### ISSUE THREAD

**Issue by codyaray** [StackStorm/st2#2815]
ST2_API_KEY and --api-key only work for some of the CLI operations.

tl;dr - 
* not supported: execution, rule, trace, trigger-instance, rule-enforcement
* supported: action, action-alias, apikey, key, policy, policy-type, run, runner, sensor, trigger, webhook

```
export ST2_API_KEY=YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ

$ st2 rule list
ERROR: 401 Client Error: Unauthorized
MESSAGE: Unauthorized - One of Token or API key required. for url: http://127.0.0.1:9101/v1/rules/?api_key=YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ&limit=50

$ st2 rule list --api-key YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ
ERROR: 401 Client Error: Unauthorized
MESSAGE: Unauthorized - One of Token or API key required. for url: http://127.0.0.1:9101/v1/rules/?api_key=YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ&limit=50

$ st2 action list
No matching items found

$ st2 action list --api-key YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ
No matching items found

$ st2 trigger list
+--------------------------------------+------+----------------------------------------------------------------------+
| ref                                  | pack | description                                                          |
+--------------------------------------+------+----------------------------------------------------------------------+

$ st2 trigger list --api-key YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ
+--------------------------------------+------+----------------------------------------------------------------------+
| ref                                  | pack | description                                                          |
+--------------------------------------+------+----------------------------------------------------------------------+
| core.st2.generic.actiontrigger       | core | Trigger encapsulating the completion of an action execution.         |

$ st2 trigger-instance list
ERROR: 401 Client Error: Unauthorized
MESSAGE: Unauthorized - One of Token or API key required. for url: http://127.0.0.1:9101/v1/triggerinstances/?api_key=YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ&limit=50

$ st2 trigger-instance list --api-key YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ
ERROR: 401 Client Error: Unauthorized
MESSAGE: Unauthorized - One of Token or API key required. for url: http://127.0.0.1:9101/v1/triggerinstances/?api_key=YzhkZjVjNWFmNmZkOWRhYjMyYzkyMzc5MTdiZDAxNGU3MmIxZWJkNTZhZGIwNTI3MWM0MGZkNmRhZmU4OTllMQ&limit=50
```

**Comment by JohnWelborn** [comment]
We are finding that run also fails.

**Comment by Kami** [comment]
Thanks again for the report and good catch.

Fix in #2831.
